src/kademlia/dht_storage.cpp

```diff
@@ -532,6 +556,36 @@ namespace
 			if (!peers.empty() && peers.capacity() / peers.size() >= 4u)
 				peers.shrink_to_fit();
 		}
+
+		void refresh_infohashes_sample()
```

Comment:
I get the impression that the code in here would be simpler if ``infohashes_sample.sample`` was ``std::vector<sha1_hash>`` instead of a string.